src/components/nvml/linux-nvml.c

```diff
@@ -315,6 +319,32 @@ getPState(nvmlDevice_t dev)
     return (unsigned long long)ret;
 }
 
+unsigned long long
+getEnergyConsumption(nvmlDevice_t dev)
+{
+    unsigned long long energy = 0;
+    nvmlReturn_t bad;
+    bad = (*nvmlDeviceGetTotalEnergyConsumptionPtr)(dev, &energy);
+
+    if (NVML_SUCCESS != bad) {
+        SUBDBG("something went wrong %s\n", (*nvmlErrorStringPtr)(bad));
+        return (unsigned long long) - 1;
+    }
+    return energy;
```

Comment:
Can you clamp this value, i.e.
```
if (energy > ULLONG_MAX) {
    energy = ULLONG_MAX
}
```
You will most likely need to include `limits.h`.